src/Main.elm

```diff
@@ -2,55 +2,91 @@ module Main exposing (..)
 
 import Browser
 import Element
+import Element exposing (rgb255)
 import Element.Input
-import Html exposing (Html, button, div, text)
+import Element.Background as Background
+import Html exposing (Html, div, text)
 import Html.Events exposing (onClick)
 
+-- STYLE
 
--- MAIN
+marigold =
+    Element.rgb255 252 186 3
+
+green =
+    Element.rgb255 8 196 27
 
+-- MAIN
 
 main =
   Browser.sandbox { init = init, update = update, view = view }
 
+{-| https://package.elm-lang.org/packages/elm/browser/latest/Browser
+    init, update, and view are defined below |-}
 
 
 -- MODEL
 
-type alias Model = Int
+type alias Counter = Int
+-- Counter is an alias for an Int (or Counter is of type Int)
 
-init : Model
+-- init is a function that returns an Int (type Model)
+init : Counter
 init =
   0
 
 
 -- UPDATE
 
-type Msg = Increment | Decrement | Reset
+type Action = Increment | Decrement | Reset
```

Comment:
Two reasons why this might not be a good idea:

1. The convention is to call this type `Msg`. Calling it `Action` will confuse other Elm programmers.
2. The semantics for how this type is used by the Elm runtime is more like a 'message' and less like an 'action'

Elaborating on the second point - 'Action' implies that your code is 'performing an action' (possibly with side-effects), which is a less accurate mental model for what happens.

See this description of the Elm Architecture: <https://guide.elm-lang.org/architecture/buttons.html#overview>

> Elm starts by rendering the initial value on screen. From there you enter into this loop:
> 
> 1. Wait for user input.
> 2. Send a message to update
> 3. Produce a new Model
> 4. Call view to get new HTML
> 5. Show the new HTML on screen
> 6. Repeat!

Later on you'll get to 'commands'. These are closer to the semantics of an action, except you're 'commanding' the Elm runtime to 'perform actions' on your behalf.

See here for more info: <https://guide.elm-lang.org/effects/>

Reply:
This will solidify in my brain eventually. :) And you are right - I was close! I was off by a few braces.